pkg/endpoint/endpoint.go

```diff
@@ -461,6 +461,11 @@ type Endpoint struct {
 	// Proxy port 0 indicates no proxy redirection.
 	// All fields within the PolicyKey and the proxy port must be in host byte-order.
 	desiredMapState PolicyMapState
+
+	///////////////////////
+	// DEPRECATED FIELDS //
```

Comment:
Nit: can you mark the code as deprecated here. Use the standard golang convention for deprecated fields:

https://blog.golang.org/godoc-documenting-go-code , search for "Deprecated".

Example: https://golang.org/src/archive/tar/common.go?h=Deprecated%3A#L175

Also specify when this field is used here.

Reply:
Thanks for the example, I didn't realise there was a convention. I'll fix this up.

I would like to put a big banner for three sections here actually, public (ensure API stability), private (if it's in this section and exported then that's erroneous), and deprecated (please please please don't use or extend). At the moment, I find it difficult to figure out exactly which fields we care about API stability for.

Reply:
Agreed, organizing the fields within the endpoint to represent their state / meaningfulness to API stability is a great idea.